plugins/fundchannel.c

```diff
@@ -326,14 +328,25 @@ static struct command_result *fundchannel_start(struct command *cmd,
 	json_out_end(ret, '}');
 	json_out_finished(ret);
 
-	/* FIXME: as a nice feature, we should check that the peer
-	 * you want to connect to is connected first. if not, we should
-	 * connect and then call fundchannel start!  */
 	return send_outreq(cmd, "fundchannel_start",
 			   fundchannel_start_done, tx_abort,
 			   fr, take(ret));
 }
 
+static struct command_result *connect_to_peer(struct command *cmd,
+                                              struct funding_req *fr)
+{
+	struct json_out *ret = json_out_new(NULL);
+
+	json_out_start(ret, NULL, '{');
+	json_out_addstr(ret, "id", node_id_to_hexstr(tmpctx, fr->id));
```

Comment:
does this work without the IP address of the client?

Reply:
Yes, it can be seen as a sanity check if the peer is connected. If it is not, this will at least try using the seed.

Reply:
yeah i think the net result of this is that it will mostly paper over how openingd disconnects when opens fail

Reply:
we should *really* fix `openingd` to not disconnect on certain classes of failure, as that cuts off the gossip with that peer. that's a much bigger fix though